tests/conftest.py

```diff
@@ -95,13 +95,25 @@ def pytest_configure(config):
     MenderTesting.set_test_conditions(config)
 
 
-def pytest_runtest_setup(item):
-    logger = log.setup_custom_logger("root", item.name)
-    logger.info("%s is starting.... " % item.name)
+# If we have xdist installed, the testlogger fixture will include the thread id
+try:
+    import xdist
+    @pytest.fixture(scope="function", autouse=True)
+    def testlogger(request, worker_id):
+        test_name = request.function.__name__
+        logger = log.setup_custom_logger("root", test_name, worker_id)
+        logger.info("%s is starting.... " % test_name)
+except ImportError:
+    @pytest.fixture(scope="function", autouse=True)
+    def testlogger(request):
+        test_name = request.function.__name__
+        logger = log.setup_custom_logger("root", test_name, None)
+        logger.info("%s is starting.... " % test_name)
+
```

Comment:
I think I am going to revert this.

I transformed it into a fixture to be able to have the xdist worker id, but the drawback is that we don't capture correctly the test name. For parametrized tests with plugin implementation `item.name` we would get `test_reboot_recovery[simulate_powerloss_in_artifact_commit_leave-test_set2]` but with fixture implementation `request.function.__name__` we get only `test_reboot_recovery`.

So unless someone illuminates me on how to get the full qualified test name in a fixture, I am reverting it.

Reply:
I found it! We can use `request.node.name`. Reworking...